src/services/generic-client.ts

```diff
@@ -0,0 +1,44 @@
+import { AxiosInstance, AxiosStatic } from 'axios';
+import { AppSession } from './app-session';
+import { RefObject } from 'react';
+
+interface ApiConnectionData {
+  jwtToken: string;
+}
+
+export class GenericClient {
```

Comment:
Good idea @kapoc3, it is nice! 

In my opinion, I think that this kind of code refactoring should be in a different PR, but I think that it is acceptable.

@cbernat @jcamposmk what do you think?

BTW, what do you think about naming it _HttpGenericClient_ or _HttpBaseClient_?

```suggestion
export class HttpBaseClient {
```

Reply:
On the other hand, in general, I prefer composition over inheritance, is there a way of doing this using composition?

Reply:
i can change inheritance for composition way, do you prefeer composition?

Reply:
In general, I prefer composition, but in this case, I am not sure if composition applies. So, if you prefer inheritance, I am fine.